### PR TITLE
Custom tslint rule: non-undefined-react-node-on-render

### PR DESCRIPTION
**Pull request by carlosmunozrodriguez**
Added a custom tslint rule with a fix to prevent returning `React.ReactNode` on `render()` method of `Component` and `PureComponent`. The fix changes the return type to `NonUndefinedReactNode`

### Recompile after modifying the rule code ###

    tsc src/rules/nonUndefinedReactNodeOnRenderRule.ts --target "ES6" --module "commonjs"

### Manually running tslint

    tslint --project .

    tslint --project . --fix

### CODE REVIEW

File: Traducir.Web/src/rules/nonUndefinedReactNodeOnRenderRule.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+export class Rule extends Lint.Rules.AbstractRule {
+    public static FailureString: string = `In Component and PureComponent, render() return type should be "NonUndefinedReactNode".`;
+
+    public apply(sourceFile: ts.SourceFile): Lint.RuleFailure[] {
+        return this.applyWithWalker(new NonUndefinedReactNodeOnRenderWalker(sourceFile, this.getOptions()));
+    }
+}
+
+// tslint:disable-next-line:max-classes-per-file
```
Comment (g3rv4) [inline Traducir.Web/src/rules/nonUndefinedReactNodeOnRenderRule.ts]: is this comment relevant to the class?

Comment (carlosmunozrodriguez) [reply Traducir.Web/src/rules/nonUndefinedReactNodeOnRenderRule.ts]: Yes, otherwise tslint itself warns about being two classes in the same file.
